src/web/lib/storage.js

```diff
@@ -0,0 +1,89 @@
+import { actions } from '../../lib/store';
+import { makeLog } from '../../lib/utils';
+
+const log = makeLog('web.storage');
+
+const THEME_STORAGE_PREFIX = 'THEME-';
+
+const themeStorageKey = key => `${THEME_STORAGE_PREFIX}${key}`;
+
+const isThemeStorageKey = key => key.indexOf(THEME_STORAGE_PREFIX) === 0;
+
+const themeKeyFromStorage = storageKey =>
+  storageKey.substr(THEME_STORAGE_PREFIX.length);
+
+export const generateThemeKey = () =>
+  `${Date.now()}-${Math.floor(Math.random() * 1000)}`;
+
+export function putTheme(key, theme) {
+  log('putTheme', key, theme);
+  const storageKey = themeStorageKey(key);
+  localStorage.setItem(
+    storageKey,
+    JSON.stringify({
```

Comment:
FWIW, trying to leave room for expansion here - e.g. if we want to store a name / description along with the theme.